ui/components/app/signature-request/signature-request.component.js

```diff
@@ -45,9 +45,72 @@ export default class SignatureRequest extends PureComponent {
       hardwareWalletRequiresConnection,
     } = this.props;
     const { address: fromAddress } = fromAccount;
-    const { message, domain = {} } = JSON.parse(data);
+    const { message, domain = {}, types, primaryType } = JSON.parse(data);
     const { metricsEvent } = this.context;
 
+    const solidityTypes = [
+      'bool',
+      'int',
+      'fixed',
+      'address',
+      'bytes',
+      'string',
+    ];
+    const sanitizeMessage = (msg, baseType) => {
+      let baseTypeType;
```

Comment:
better name for this `baseTypeType` ?

Reply:
Maybe `baseTypeDefinition`?

Reply:
And also the below variable `nestedTypeType` could be `nestedTypeDefinition`